gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -91,18 +94,16 @@
   private final RowLevelPolicyCheckResults rowLevelPolicyCheckingResult;
 
   // A bounded blocking queue in between the parent task and this fork
-  private final BoundedBlockingRecordQueue<Object> recordQueue;
+  private final BlockingQueue<Object> recordQueue;
 
   private final Closer closer = Closer.create();
 
   // The writer will be lazily created when the first data record arrives
   private Optional<DataWriter<Object>> writer = Optional.absent();
 
-  // This is used by the parent task to signal that it has done pulling records and this fork
-  // should not expect any new incoming data records. This is written by the parent task and
-  // read by this fork. Since this flag will be updated by only a single thread and updates to
-  // a boolean are atomic, volatile is sufficient here.
-  private volatile boolean parentTaskDone = false;
+  // Submitted to the recordQueue by the parent task to indicated that no
+  // more records are to be expected
+  private static final Object PARENTTASKDONE = new Object();
```

Comment:
Can you change the variable name to `PARENT_TASK_DONE` in order to be consistent with the Gobblin coding style: http://gobblin.readthedocs.org/en/latest/developer-guide/CodingStyle/